nbconvert/filters/markdown_mistune.py

```diff
@@ -21,6 +21,12 @@
 from nbconvert.filters.strings import add_anchor
 
 
+class MathBlockGrammar(mistune.BlockGrammar):
+    block_math = re.compile(r"^\$\$(.*?)\$\$|^\\\\\[(.*?)\\\\\]", re.DOTALL)
```

Comment:
As described in my comment, this should be including the delimiters in its content to be passed through to the inline parser.

Reply:
It also would make sense to give this rule a new name, since this conflicts with the current meaning of `block_math` in the `InlineLexer`. One possibility would be `multiline_math`.